File: src/main/数组/t989/AddToArrayForm.java
package 数组.t989;

import java.util.LinkedList;
import java.util.List;

/**
 * @author: sotowang
 * @date: 2021/01/22 17:02
 */
public class AddToArrayForm {
    public List<Integer> addToArrayForm(int[] A, int K) {
        int len = A.length;
        int lastNum =K;
        LinkedList<Integer> ret= new LinkedList<>();

        int i = len-1;
        while(i >=0 || lastNum > 0){
            if(i >= 0){
                lastNum+=A[i];
            }
            ret.addFirst(lastNum%10);
            lastNum/=10;
            i--;
        }

        return ret;
    }
}

File: src/main/数组/t989/addToArrayForm.go
package main

import "fmt"

func addToArrayForm(A []int, K int) []int {
	n := len(A)
	lastNum := K
	res := make([]int, 0)
	i := n - 1
	for i >= 0 || lastNum > 0 {
		if i >= 0 {
			lastNum += A[i]
		}
		res = append(res, lastNum%10)
		lastNum /= 10
		i--
	}
	reverse(res)
	return res
}
func reverse(res []int) {
	n := len(res)
	for i := 0; i < n/2; i++ {
		res[i], res[n-i-1] = res[n-i-1], res[i]
	}
}

func main() {
	A := []int{9, 9, 9, 9}
	fmt.Println(addToArrayForm(A, 1))
}
